packages/react-text/src/components/Text/Text.types.ts

```diff
@@ -1,99 +1,90 @@
-import * as React from 'react';
-import type { ComponentPropsCompat, ComponentStateCompat } from '@fluentui/react-utilities';
+import type { ComponentProps, ComponentState, IntrinsicShorthandProps } from '@fluentui/react-utilities';
 
 /**
- * Text Props
+ * Text slots
  */
-export interface TextProps extends ComponentPropsCompat, React.HTMLAttributes<HTMLElement> {
+export type TextSlots = {
+  root: IntrinsicShorthandProps<'span', 'p' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'pre'>;
+};
+
+export interface TextCommons {
   /**
    * Wraps the text content on white spaces.
    *
    * @defaultValue true
    */
-  wrap?: boolean;
+  wrap: boolean;
 
   /**
    * Truncate overflowing text for block displays.
    *
    * @defaultValue false
    */
-  truncate?: boolean;
+  truncate: boolean;
 
   /**
    * Applies a block display for the content.
    *
    * @defaultValue false
    */
-  block?: boolean;
+  block: boolean;
 
   /**
    * Applies the italic font style to the content.
    *
    * @defaultValue false
    */
-  italic?: boolean;
+  italic: boolean;
 
   /**
    * Applies the underline text decoration to the content.
    *
    * @defaultValue false
    */
-  underline?: boolean;
+  underline: boolean;
 
   /**
    * Applies the strikethrough text decoration to the content.
    *
    * @defaultValue false
    */
-  strikethrough?: boolean;
+  strikethrough: boolean;
 
   /**
    * Applies font size and line height based on the theme tokens.
    *
    * @defaultValue 300
    */
-  size?: 100 | 200 | 300 | 400 | 500 | 600 | 700 | 800 | 900 | 1000;
+  size: 100 | 200 | 300 | 400 | 500 | 600 | 700 | 800 | 900 | 1000;
 
   /**
    * Applies the font family to the content.
    *
    * @defaultValue base
    */
-  font?: 'base' | 'monospace' | 'numeric';
+  font: 'base' | 'monospace' | 'numeric';
 
   /**
    * Applies font weight to the content.
    *
    * @defaultValue regular
    */
-  weight?: 'regular' | 'medium' | 'semibold';
+  weight: 'regular' | 'medium' | 'semibold';
 
   /**
    * Aligns text based on the parent container.
    *
    * @defaultValue start
    */
-  align?: 'start' | 'center' | 'end' | 'justify';
-
-  /**
-   * Component to be rendered as.
-   *
-   * @defaultValue span
-   */
-  as?: 'span' | 'p' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'pre';
+  align: 'start' | 'center' | 'end' | 'justify';
 }
 
 /**
- * Names of TextProps that have a default value in useText
+ * Text Props
  */
-export type TextDefaultedProps = never;
+export type TextProps = ComponentProps<TextSlots> & Partial<TextCommons>;
 
 /**
  * State used in rendering Text
  */
-export interface TextState extends ComponentStateCompat<TextProps, TextDefaultedProps> {
-  /**
-   * Ref to the root element
-   */
-  ref: React.Ref<HTMLElement>;
-}
+export interface TextState extends ComponentState<TextSlots>, TextCommons {}
```

Comment:
AFAIK the `interface` should be changed to `type` based on discussions and the RFC found here: https://github.com/microsoft/fluentui/issues/19823 and PR: https://github.com/microsoft/fluentui/pull/19850. WDYT?

Reply:
Good point, I've gone ahead and migrated all interfaces to types.